pkgs/top-level/qt-packages.nix

```diff
@@ -0,0 +1,171 @@
+
+# Qt libraries set.
```

Comment:
This says the file is for Qt libraries, but it includes all of `plasma5` and `kdeApplications`. Why isn't this all Qt packages? I think that would be a better solution to the version-inconsistency problem.

Reply:
I am not sure whether we should use it only for libraries, or also applications. When I created this I aimed for libraries only.

The issue we have here seems similar to Python, however, somewhat simpler because we do not have to deal with various versions of libraries other than the versions of Qt. This is the reason Python applications are outside of `python-packages.nix`, so you can use different versions. But with Qt I *think* we do not need it, so then we can indeed put both applications and libraries in the same file.

Reply:
The distinction between an application and a library can't be guaranteed to be clear for ever. In #101369 I learned that some "applications" from `kdeApplications` were used as inputs to other. Hence I support the idea to put all qt applications there (docs should be updated as well).